drivers/bluetooth/hci/userchan.c

```diff
@@ -257,14 +298,16 @@ SYS_INIT(bt_uc_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
 
 static void cmd_bt_dev_found(char *argv, int offset)
 {
-	if (strncmp(&argv[offset], "hci", 3) || strlen(&argv[offset]) < 4) {
-		posix_print_error_and_exit("Error: Invalid Bluetooth device "
-					   "name '%s' (should be e.g. hci0)\n",
-					   &argv[offset]);
-		return;
+	arg_found = true;
+	if (strncmp(&argv[offset], "hci", 3) == 0 && strlen(&argv[offset]) >= 4) {
+		/* It's a hciX string */
+		hci_socket = true;
+		bt_dev_index = strtol(&argv[offset + 3], NULL, 10);
+	} else if (sscanf(&argv[offset], "%15[^:]:%d", ip_addr, &port) == 2) {
+		hci_socket = false;
+	} else {
+		invalid_argument = true;
```

Comment:
This variable is never used. Would it not be better to print an error here directly?

Reply:
Do you mean just an LOG_ERR  or already exit with posix_print_error_and_exit

Reply:
`posix_print_error_and_exit()`.  The logging subsystem is not yet up, and anyhow, the best we can do is stop right away telling the user what the problem is. There is no point continuing.